service/src/main/java/uk/nhs/adaptors/gp2gp/mhs/exception/UnrecognisedInteractionIdException.java

```diff
@@ -0,0 +1,10 @@
+package uk.nhs.adaptors.gp2gp.mhs.exception;
+
+public class UnrecognisedInteractionIdException extends RuntimeException {
+
+    private static final String EXCEPTION_MESSAGE = "Received an unrecognized %s message with conversation_id: %s";
```

Comment:
Just having a think about this one, say that `RCMR_MT030101UK05` was an unknown Interaction ID for illustrations sake,  then this message would read:

`Received an unrecognized RCMR_MT030101UK05 message with conversation_id: 274a7a48-d48a-46dd-9da3-34673cc409d3`

Based on the name of the exception message, it may be worth highlighting the point that it's an unrecongised Interaction ID, for example:

`Received a message with an unrecognized Interaction ID (RCMR_MT030101UK05), Conversation ID: 274a7a48-d48a-46dd-9da3-34673cc409d3`

Reply:
I think we just work with the types like Continue, ACK etc which should be pretty close to what you are pointing out:
Received an unrecognized Continue message with conversation_id: 274a7a48-d48a-46dd-9da3-34673cc409d3